tests/robot-tests/tests/public_api/public_api_dataset_cancel_and_removal.robot

```diff
@@ -0,0 +1,142 @@
+
+
+    user checks table column heading contains    1    1    Draft version    xpath://table[@data-testid='draft-api-data-sets']
+    user checks table column heading contains    1    2    Name             xpath://table[@data-testid='draft-api-data-sets']
+    user checks table column heading contains    1    3    Status           xpath://table[@data-testid='draft-api-data-sets']
+    user checks table column heading contains    1    4    Actions          xpath://table[@data-testid='draft-api-data-sets']
+
+
+    user checks table cell contains    1    1    v1.0    xpath://table[@data-testid='draft-api-data-sets']
+    user checks table cell contains    1    3    Ready    xpath://table[@data-testid='draft-api-data-sets']
+
+Click on 'cancel' button while attempting to remove draft API dataset
+
+    user clicks button in table cell    1    4    Remove draft    xpath://table[@data-testid='draft-api-data-sets']
+
+    ${modal}=    user waits until modal is visible     Remove this draft API data set version
+    User Clicks Button By Indexing    Cancel    3
+    user waits until h2 is visible    API data sets
+
+Verify the contents inside the 'Draft API datasets' table
+    user waits until h3 is visible    Draft API data sets
+
+
+    user checks table column heading contains    1    1    Draft version    xpath://table[@data-testid='draft-api-data-sets']
+    user checks table column heading contains    1    2    Name             xpath://table[@data-testid='draft-api-data-sets']
+    user checks table column heading contains    1    3    Status           xpath://table[@data-testid='draft-api-data-sets']
+    user checks table column heading contains    1    4    Actions          xpath://table[@data-testid='draft-api-data-sets']
+
+
+    user checks table cell contains    1    1    v1.0    xpath://table[@data-testid='draft-api-data-sets']
+    user checks table cell contains    1    3    Ready    xpath://table[@data-testid='draft-api-data-sets']
+
+Remove draft API dataset
+
+    user clicks button in table cell    1    4    Remove draft    xpath://table[@data-testid='draft-api-data-sets']
+    
+    ${modal}=    user waits until modal is visible     Remove this draft API data set version
+    user clicks button     Remove this API data set version
+    user waits until h2 is visible    API data sets
+
+Add headline text block to Content page
+    user navigates to content page    ${PUBLICATION_NAME}
+    user adds headlines text block
+    user adds content to headlines text block    Headline text block text
+
+Approve first release
+    user clicks link    Sign off
+    user approves release for immediate publication
+
+
+
+
+
+*** Keywords ***
+Verify status of API Datasets
+    [Arguments]    ${expected_status}
+    user waits for caches to expire
+    ${status_value}=    get text    xpath:(//div[@data-testid="Status"]//dd[@data-testid="Status-value"]//strong)[2]
+    should be equal as strings    ${status_value}    ${expected_status}
```

Comment:
I think you create the same keyword at the bottom of `public_api_restricted.robot` in the EES-5409 PR.

I'd pull this out into a admin-public-api.robot file and have both test suites access the `Verify status of API datasets` keyword from there, as I imagine this keyword will be used in other Public API UI tests.